### PR TITLE
add btree Ascend、Descend method and unitest.

### PR DESCRIPTION
**Pull request by weijiew**

### CODE REVIEW

File: index/btree.go [hunk 1]
```diff
@@ -64,3 +64,23 @@ func (mt *MemoryBTree) Delete(key []byte) (*wal.ChunkPosition, bool) {
 func (mt *MemoryBTree) Size() int {
 	return mt.tree.Len()
 }
+
+// Ascend calls f for each item in the tree in ascending order. If f returns false, iteration stops.
```
Comment (roseduan) [inline index/btree.go:68]: `f -> handleFunc/handleFn` maybe more clear?

The parameter of the function is not ChunkPosition, but the value in the file, we must read the value according to the position.

Comment (weijiew) [reply index/btree.go]: Based on your suggestion, I have made further modifications to the code. Could you please review it to see if it meets your expectations? If not, please point out any issues, and I will make further adjustments. Due to my daytime commitments for company projects, I can only work on the open-source project during the evenings. I apologize for any delay in responding.

File: db.go [hunk 2]
```diff
@@ -265,6 +265,30 @@ func (db *DB) Watch() (chan *Event, error) {
 	return db.watchCh, nil
 }
 
+// Ascend calls handleFn for each item in the tree in ascending order.
```
Comment (roseduan) [inline db.go:268]: in the tree -> in the db

File: index/btree.go (hunk 1 above)
Comment (roseduan) [reply index/btree.go]: it`s so kind of you to tell me this, I also maintain the project in my spare time, so just feel free to contribute.

Comment (weijiew) [reply index/btree.go]: Thank you very much for the project you've written, which has enabled me to further study the Bitcask model.

File: db.go (hunk 2 above)
Comment (weijiew) [reply db.go]: ok

File: db.go [hunk 3]
```diff
@@ -265,6 +265,30 @@ func (db *DB) Watch() (chan *Event, error) {
...
+func (db *DB) Ascend(handleFn func(k []byte, v []byte) bool) error {
+	db.index.Ascend(func(key []byte, pos *wal.ChunkPosition) bool {
```
[the remark this answers is not in the document]
Comment (weijiew) [reply db.go]: Thank you for your guidance. I have made additional modifications. Could you please review it once more and let me know if there are any further changes needed?

**Comment by roseduan** [comment]
Thanks

**Pull request merged**